src/main/java/reactor/core/publisher/FluxSink.java

```diff
@@ -60,11 +62,27 @@
 
 	/**
 	 * Ensures that calls to next, error and complete are properly serialized.
+	 * {@link Flux#create(java.util.function.Consumer)} or
+	 * {@link Flux#create(java.util.function.Consumer, OverflowStrategy)} can be
+	 * used to create a serialized {@link FluxSink}.
 	 * @return the serialized {@link FluxSink}
 	 */
+	@Deprecated
 	FluxSink<T> serialize();
 
 	/**
+	 * Attaches a {@link LongConsumer} to this {@link FluxSink} that will be notified of
+	 * any request to this sink. This method is supported only on serialized sinks created
```

Comment:
@dfeist Thank you for the review. After discussion with @smaldini I have updated the PR to handle `onRequest` for push-only sinks as well. For push-only `FluxSink`, the `onRequest` consumer is invoked once within `onRequest` with a request of `Long.MAX_VALUE` avoiding serialization issues. This enables similar code to be used for handling both push-only and push-pull sinks.